kyaml/kio/byteio_writer.go

```diff
@@ -134,6 +135,25 @@ func (w ByteWriter) Write(inputNodes []*yaml.RNode) error {
 	return encoder.Encode(doc)
 }
 
+// setSeqIndent reads the SeqIndentAnnotation from node and sets the sequence indentation
+// in the encoder
+// if the resource doesn't have SeqIndentAnnotation it will use CompactSeqIndent
+func setSeqIndent(node *yaml.RNode, encoder *yaml.Encoder) {
```

Comment:
This function is doing three things:
1. Reading the seqIndentType from a resource
2. Setting the encoder with new seqIndentType
3. Clearing the indent Annotation and also empty annotations etc.

I have suggested another approach to decouple these which will also make the new code consistent when it comes to reading, clearing the annotations.

Reply:
Renamed the method to make it more descriptive with updated functionality.